File: r_package/Rdnn/R/env.R


dnn.env = function () {
	dnn_env = Sys.getenv("DNN_HOME", "/usr/local/dnn")	
	if(!file.exists(dnn_env)) {
		stop("Can't find dnn home, make sure it is installed and environment varibale DNN_HOME points at a right place, or it is installed in system path /usr/local/dnn")
	}
	return(dnn_env)
}

ts.path = function(...) {
	file.path(dnn.env(), "ts", ...)
}

spikes.path = function(...) {
	file.path(dnn.env(), "spikes", ...)
}

ds.path = function(...) {
	file.path(dnn.env(), "datasets", ...)
}

scripts.path = function(...) {
    file.path(dnn.env(), "r_scripts", ...)
}

File: scripts/lib/env.py

import os
from os.path import join as pj

DNN_HOME = os.environ.get("DNN_HOME")

if DNN_HOME is None:
    raise Exception("Can't find DNN_HOME environment variable. Check that dnn is installed")


runs_dir = pj(DNN_HOME, "runs")
r_scripts_dir = pj(DNN_HOME, "r_scripts")
spikes_dir = pj(DNN_HOME, "spikes")
lib_dir = pj(DNN_HOME, "lib")
include_dir = pj(DNN_HOME, "include")
